src/main/java/org/assertj/core/api/AbstractMapAssert.java

```diff
@@ -396,6 +396,180 @@ public SELF hasEntrySatisfying(K key, Condition<? super V> valueCondition) {
+    maps.assertHasEntrySatisfying(info, actual, entryCondition);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual map contains an entry with key satisfying {@code keyCondition} and value satisfying {@code valueCondition}.
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * Map&lt;TolkienCharacter, Ring&gt; ringBearers = new HashMap&lt;&gt;();
+   * ringBearers.put(galadriel, nenya);
+   * ringBearers.put(gandalf, narya);
+   * ringBearers.put(elrond, vilya);
+   * ringBearers.put(frodo, oneRing);
+   *
+   * Condition&lt;TolkienCharacter&gt; isMan =
+   *   new Condition&lt;TolkienCharacter&gt;("is man") {
+   *     public boolean matches(TolkienCharacter value) {
+   *       return value.getRace() == MAN;
+   *     }
+   *   };
+   *
+   * Condition&lt;Ring&gt; oneRingBearer =
+   *   new Condition&lt;Ring&gt;("One ring bearer") {
+   *     public boolean matches(Ring value) {
+   *       return value == oneRing;
+   *     }
+   *   };
+   *
+   * // assertion will fail
+   * assertThat(ringBearers).hasEntrySatisfying(isMan, oneRingBearer);
+   *
+   * ringBearers.put(isildur, oneRing);
+   *
+   * // now assertion will pass
+   * assertThat(ringBearers).hasEntrySatisfying(isMan, oneRingBearer);
+   * </code></pre>
+   *
+   * @param keyCondition the condition for entry key.
+   * @param valueCondition the condition for entry value.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if any of the given conditions is {@code null}.
+   * @throws AssertionError if the actual map is {@code null}.
+   * @throws AssertionError if there is no entry matching given {@code keyCondition} and {@code valueCondition}.
+   * @since 2.8.0
+   */
+  public SELF hasEntrySatisfying(Condition<? super K> keyCondition, Condition<? super V> valueCondition) {
+    maps.assertHasEntrySatisfyingConditions(info, actual, keyCondition, valueCondition);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual map contains an entry with key satisfying {@code keyCondition}.
+   * <p>
+   * Example:
+   * <pre><code class='java'>
+   * Map&lt;TolkienCharacter, Ring&gt; ringBearers = new HashMap&lt;&gt;();
+   * ringBearers.put(galadriel, nenya);
+   * ringBearers.put(gandalf, narya);
+   * ringBearers.put(elrond, vilya);
+   * ringBearers.put(frodo, oneRing);
+   *
+   * Condition&lt;TolkienCharacter&gt; isMan =
+   *   new Condition&lt;TolkienCharacter&gt;("is man") {
+   *     public boolean matches(TolkienCharacter value) {
+   *       return value.getRace() == MAN;
+   *     }
+   *   };
+   *
+   * Condition&lt;TolkienCharacter&gt; isOrc =
+   *   new Condition&lt;TolkienCharacter&gt;("is orc") {
+   *     public boolean matches(TolkienCharacter value) {
+   *       return value.getRace() == ORC;
+   *     }
+   *   };
+   *
+   * // assertion will fail
+   * assertThat(ringBearers).hasKeySatisfying(isOrc);
+   *
+   * // assertion will pass
+   * assertThat(ringBearers).hasKeySatisfying(isMan);
```

Comment:
that actually fails as galadriel and elrond are elves, frodo is a hobbit and gandalf is not a man but a maia :smile_cat: